### PR TITLE
Add option to set decontamination database index path

### PR DESCRIPTION
**Pull request by mbhall88**
Addresses https://github.com/mbhall88/tbpore/issues/34#issuecomment-1271666527.

Also, update mykrobe to v0.12 as this uses the newest panel that includes the WHO catalogue

### CODE REVIEW

**Review by leoisl — APPROVED** [review]
All fine! I just updated the README.md with a line saying there is no need to move the DB to the default location and with the new CLI option

**Comment by leoisl** [comment]
Argh, just noticed the decontamination DB is composed of two files, the large minimap2 index and the metadata about the indexed genomes ([remove_contam.tsv.gz](https://github.com/mbhall88/tbpore/blob/main/data/decontamination_db/remove_contam.tsv.gz)). So if we want to be precise, the `--db` option should point to a dir with these two files.

If the user specified a different decontamination index (not just a different location, an actual different index), then `data/decontamination_db/remove_contam.tsv.gz` might not have the correct metadata for the genomes in this different index. It would be even complicated to update: the user would need to update this metadata file `data/decontamination_db/remove_contam.tsv.gz`, while the index lies somewhere else, and I think this update might not be possible in a container? However, I don't think we have this use case now, we are always using the same decontamination DB, but the current `--db` option implementation introduce this bug if a different index is used...

**Comment by mbhall88** [comment]
True....

We could also add a metadata option? And just document that if the decontamination index you're using is not the one downloaded from us, then you should also provide a metadata file?

File: tbpore/tbpore.py [hunk 1]
```diff
@@ -219,6 +226,22 @@ def process(
...
+            "but are using the default metadata file. If you are using the tbpore-provided "
+            "database, this is fine. Otherwise we recommend creating your own metadata "
+            "file"
+        )
+
+    if metadata:
+        if not db:
+            logger.info(
+                "You have specified a custom metadata file, but the default "
+                "decontamination database. We trust you know what you're doing"
```
Comment (leoisl) [inline tbpore/tbpore.py:241]: > We trust you know what you're doing

:joy:

**Review by leoisl — APPROVED** [review]
Thanks for this, looks great! There is indeed still some improvements to be made, maybe specify the metadata in the fasta header of the reference to be indexed by `minimap2`, and thus the `minimap2` index includes both the data and metadata, or a `.tar` file that combines both, but we do provide the ways to change both the DB and the metadata, although we trust the user to be careful if the DB is changed. Anyway, in practice our use case is just changing the location of the default DB, so this works perfectly!

**Pull request merged**